tests/ui/rfc-2497-if-let-chains/ensure-that-let-else-does-not-interact-with-let-chains.stderr

```diff
@@ -37,6 +37,11 @@ help: add a block here
    |
 LL |     if let Some(n) = opt else {
    |                         ^
+help: There may be an extra `if` before the `let...else`
+  --> $DIR/ensure-that-let-else-does-not-interact-with-let-chains.rs:24:5
+   |
+LL |     if let Some(n) = opt else {
+   |     ^^
```

Comment:
Related test source code: [rfc-2497-if-let-chains/ensure-that-let-else-does-not-interact-with-let-chains.rs](https://github.com/rust-lang/rust/blob/cf2dff2/tests/ui/rfc-2497-if-let-chains/ensure-that-let-else-does-not-interact-with-let-chains.rs).

This behavior is correct as this hint should only appear when it is ambiguous if the user is try to use an `if-let` or an `let..else`.

The other two cases that this test handles are:
```rust
    if let Some(n) = opt && n == 1 else {
    //~^ ERROR this `if` expression is missing a block after the condition
        return;
    };
    if let Some(n) = opt && let another = n else {
    //~^ ERROR this `if` expression is missing a block after the condition
        return;
    };
```

which are clear attempts at `if-let` chains.

This also indicates that we don't emit this help on `while-let` -- this wasn't in scope of the original issue so I took a conservative approach in this PR.